src/System.Management.Automation/engine/Subsystem/DscSubsystem/ICrossPlatformDsc.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using System.Management.Automation.Internal;
+using System.Management.Automation.Language;
+
+namespace System.Management.Automation.Subsystem
+{
+    /// <summary>
+    /// Interface for implementing a cross platform desired state configuration component.
+    /// </summary>
+    public interface ICrossPlatformDsc : ISubsystem
+    {
+        /// <summary>
+        /// Subsystem kind.
+        /// </summary>
+        SubsystemKind ISubsystem.Kind => SubsystemKind.CrossPlatformDsc;
+
+        /// <summary>
+        /// Dsc initializer function.
+        /// </summary>
+        void LoadDefaultCimKeywords(Collection<Exception> errors);
+
+        /// <summary>
+        /// Clear internal class caches.
+        /// </summary>
+        void ClearCache();
+
+        /// <summary>
+        /// Flag that shows whether v2 or v3 API was initialized.
+        /// </summary>
+        bool NewApiIsUsed { get; }
```

Comment:
If the DSC subsystem is shipped with `PSDesiredStateConfiguration` v3, then we know that the v3 version of `PSDesiredStateConfiguration` must be imported for the DSC subsystem implementation to be registered, right? Then we know it will use the new APIs for sure, I guess.